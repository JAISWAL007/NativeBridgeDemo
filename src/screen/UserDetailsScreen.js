import React, {useState} from 'react';
import {Text, TextInput, StyleSheet, TouchableOpacity} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import {userAction} from '../redux/actions';
import {useDispatch} from 'react-redux';
import Strings from '../Constant';

const UserDetailsScreen = ({navigation}) => {
  const [userName, setUserName] = useState('');
  const dispatch = useDispatch();

  const onPress = () => {
    if (userName) {
      dispatch(userAction(userName));
      navigation.navigate('PostLogin', {
        screen: 'Dashboard',
      });
    } else {
      alert(Strings.PLEASE_ENTER_NAME);
    }
  };

  const validateUserName = (name) => {
    if (name) {
      return setUserName(name);
    }
    setUserName('');
  };

  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.titleStyle}>{Strings.USER_NAME}</Text>
      <TextInput
        onChangeText={validateUserName}
        placeholder={Strings.ENTER_USER_NAME}
        style={styles.inputStyle}
        returnKeyType="done"
      />
      <TouchableOpacity onPress={onPress} style={styles.wrapperCustom}>
        <Text style={styles.buttonTitleStyle}>{Strings.NEXT}</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 20,
    alignItems: 'center',
    backgroundColor: 'gray',
    justifyContent: 'center',
    paddingBottom: 50,
  },
  titleStyle: {
    fontSize: 28,
    fontWeight: 'bold',
    marginBottom: 40,
  },
  inputStyle: {
    height: 50,
    borderWidth: 0,
    width: '100%',
    marginBottom: 10,
    paddingLeft: 20,
    borderColor: '#C7F5F3',
    borderRadius: 5,
    backgroundColor: '#fff',
  },
  buttonTitleStyle: {
    paddingVertical: 10,
    paddingHorizontal: 20,
    borderRadius: 10,
    fontWeight: 'bold',
    color: '#fff',
  },
  wrapperCustom: {
    borderRadius: 8,
    width: '100%',
    alignItems: 'center',
    padding: 6,
    backgroundColor: '#0ED814',
    marginBottom: 20,
  },
});

export default UserDetailsScreen;
