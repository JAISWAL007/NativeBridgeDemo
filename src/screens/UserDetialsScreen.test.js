import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import UserDetailsScreen from './UserDetailsScreen';
import {Provider} from 'react-redux';
import {Alert} from 'react-native';
import store from '../redux/store';

describe('User Details Screen', () => {
  const mockNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };

  it('Should render User Details Screen ', () => {
    const tree = render(
      <Provider store={store}>
        <UserDetailsScreen navigation={mockNavigation} />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should show error message when user without enter input field hit next button ', async () => {
    jest.spyOn(Alert, 'alert');
    const {getByTestId} = render(
      <Provider store={store}>
        <UserDetailsScreen navigation={mockNavigation} />,
      </Provider>,
    );
    const input = getByTestId('input-field');
    await fireEvent(input, 'onChangeText', '');
    await fireEvent.press(getByTestId('button'));
    expect(Alert.alert).toHaveBeenCalledWith('Error', 'please enter name', [
      {text: 'Ok'},
    ]);
  });

  it('Should able to navigate to home screen when user click on home button', async () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <UserDetailsScreen navigation={mockNavigation} />,
      </Provider>,
    );
    const input = getByTestId('input-field');
    await fireEvent(input, 'onChangeText', 'Testing');
    await fireEvent.press(getByTestId('button'));
    expect(mockNavigation.navigate).toBeCalledTimes(1);
  });
});
