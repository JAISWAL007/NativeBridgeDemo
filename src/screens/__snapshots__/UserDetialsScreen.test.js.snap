// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Details Screen Should render User Details Screen  1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "gray",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 50,
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 28,
        "fontWeight": "bold",
        "marginBottom": 40,
      }
    }
  >
    User Name
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter user name"
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#C7F5F3",
        "borderRadius": 5,
        "borderWidth": 0,
        "height": 50,
        "marginBottom": 10,
        "paddingLeft": 20,
        "width": "100%",
      }
    }
    testID="input-field"
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0ED814",
        "borderRadius": 8,
        "marginBottom": 20,
        "opacity": 1,
        "padding": 6,
        "width": "100%",
      }
    }
    testID="button"
  >
    <Text
      style={
        Object {
          "borderRadius": 10,
          "color": "#fff",
          "fontWeight": "bold",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      Next
    </Text>
  </View>
</RNCSafeAreaView>
`;
